commit 664d190d219b6f20063058f6ecdc950cbe54716e
Author: dann frazier <dannf@hp.com>
Date:   Sun Feb 14 13:30:59 2010 -0700

    commit 6f5a55f1a6c5abee15a0e878e5c74d9f1569b8b0
    Author: Linus Torvalds <torvalds@linux-foundation.org>
    Date:   Fri Feb 5 16:16:50 2010 -0800
    
       Fix potential crash with sys_move_pages
    
    [Adjusted to apply to Debian's 2.6.24]

diff --git a/mm/migrate.c b/mm/migrate.c
index 9c89275..a552f4b 100644
--- a/mm/migrate.c
+++ b/mm/migrate.c
@@ -998,6 +998,9 @@ SYSCALL_DEFINE6(move_pages, pid_t, pid, unsigned long, nr_pages,
 				goto out;
 
 			err = -ENODEV;
+			if (node < 0 || node >= MAX_NUMNODES)
+				goto out;
+
 			if (!node_state(node, N_HIGH_MEMORY))
 				goto out;
 
